### PR TITLE
store/tikv: Continue GC when error happens during UnsafeDestroyRange

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

### What problem does this PR solve? 

This PR makes GC continues working when sending some UnsafeDestroyRange requests returns error.

Also, a metrics is added to monitor failures of UnsafeDestroyRanges

### What is changed and how it works?

(see above)

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the `tidb-ansible` repository

### CODE REVIEW

**Comment by shenli** [comment]
There is no unit test in this PR.

File: metrics/gc_worker.go [hunk 1]
```diff
@@ -67,4 +67,12 @@ var (
 			Name:      "gc_region_too_many_locks",
 			Help:      "Counter of gc scan lock request more than once in the same region.",
 		})
+
+	GCUnsafeDestroyRangeFailuresCounter = prometheus.NewCounter(
```
Comment (jackysp) [inline metrics/gc_worker.go]: Need to register it.

**Comment by MyonKeminta** [comment]
@shenli @jackysp PTAL again

File: metrics/gc_worker.go (hunk 1 above)
Comment (zhangjinpeng1987) [reply metrics/gc_worker.go]: done

**Comment by MyonKeminta** [comment]
It seems some tests failed.

**Comment by MyonKeminta** [comment]
/run-all-tests
@zhangjinpeng1987 @disksing PTAL

File: store/tikv/gcworker/gc_worker.go [hunk 2]
```diff
@@ -624,27 +642,51 @@ func (w *GCWorker) sendUnsafeDestroyRangeRequest(ctx context.Context, startKey [
...
+		if err1 != nil {
+			err = err1
+		}
+	}
+
 	wg.Wait()
 
-	return errors.Trace(err)
+	if err != nil {
+		return errors.New("[gc worker] destroy range failed on some stores")
```
Comment (lonng) [inline store/tikv/gcworker/gc_worker.go]: Prefer to user `errors.Annotatef` to keep the original error message.

**Comment by MyonKeminta** [comment]
@disksing PTAL again thanks!

**Comment by MyonKeminta** [comment]
Oh, sorry, please wait a minute

File: store/tikv/gcworker/gc_worker.go [hunk 3]
```diff
@@ -546,14 +546,23 @@ func (w *GCWorker) deleteRanges(ctx context.Context, safePoint uint64) error {
...
+				zap.Error(err))
+			continue
 		}
 
 		se := createSession(w.store)
 		err = util.CompleteDeleteRange(se, r)
 		se.Close()
 		if err != nil {
-			return errors.Trace(err)
+			logutil.Logger(ctx).Info("[gc worker] failed to mark delete range task done",
```
Comment (lonng) [inline store/tikv/gcworker/gc_worker.go]: Do we need to change the `GCUnsafeDestroyRangeFailuresCounter` to the type `CounterVec` and add a label (e.g: "send" and "save") to indicate the stage of failure occurred.

File: store/tikv/gcworker/gc_worker.go (hunk 2 above)
Comment (MyonKeminta) [reply store/tikv/gcworker/gc_worker.go]: These errors were already logged so I thought it's ok to drop them... And there may be multiple regions returned from above, so I think collect them into an `[]error` and use `errors.Errorf`. It's that ok?

File: store/tikv/gcworker/gc_worker.go (hunk 3 above)
Comment (MyonKeminta) [reply store/tikv/gcworker/gc_worker.go]: Good idea

File: store/tikv/gcworker/gc_worker.go (hunk 2 above)
Comment (lonng) [reply store/tikv/gcworker/gc_worker.go]: The error log will be scattered throughout the file in the current implementation. I think we should collect all error messages in a GC job and log them after the job finishes.

**Comment by MyonKeminta** [comment]
@lonng PTAL again. Is that ok?

**Comment by MyonKeminta** [comment]
@disksing PTAL thanks!

File: store/tikv/gcworker/gc_worker.go [hunk 4]
```diff
@@ -528,14 +528,24 @@ func (w *GCWorker) deleteRanges(ctx context.Context, safePoint uint64, concurren
...
 		se := createSession(w.store)
 		err = util.CompleteDeleteRange(se, r)
 		se.Close()
 		if err != nil {
-			return errors.Trace(err)
+			logutil.Logger(ctx).Error("[gc worker] failed to mark delete range task done",
+				zap.String("uuid", w.uuid),
+				zap.Binary("startKey", startKey),
+				zap.Binary("endKey", endKey),
+				zap.Error(err))
```
Comment (AndreMouche) [inline store/tikv/gcworker/gc_worker.go]: So we wouldn't return any error when complete delete range ffailed?

**Comment by disksing** [comment]
LGTM

**Comment by MyonKeminta** [comment]
@disksing Could you please give me a approve?

**Pull request merged**